web/service/src/main/java/org/mobilitydata/gtfsvalidator/web/service/util/StorageHelper.java

```diff
@@ -0,0 +1,166 @@
+package org.mobilitydata.gtfsvalidator.web.service.util;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.cloud.storage.*;
+import java.io.*;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationContext;
+
+/** Helper class for interacting with GCS. */
+public class StorageHelper {
+  static final String JOB_INFO_BUCKET_NAME = "gtfs-validator-results";
+  static final String JOB_FILENAME_PREFIX = "job";
+  static final String JOB_FILENAME_SUFFIX = ".json";
+  static final String JOB_FILENAME = JOB_FILENAME_PREFIX + JOB_FILENAME_SUFFIX;
+  public static final String TEMP_FOLDER_NAME = "gtfs-validator-temp";
+  static final String USER_UPLOAD_BUCKET_NAME = "gtfs-validator-user-uploads";
+  static final String RESULTS_BUCKET_NAME = "gtfs-validator-results";
+  static final String FILE_NAME = "gtfs-job.zip";
+
+  private final Logger logger = LoggerFactory.getLogger(StorageHelper.class);
+  private Storage storage;
+  private ApplicationContext applicationContext;
+
+  // constructor
+  public StorageHelper(Storage storage, ApplicationContext applicationContext) {
+    this.storage = storage;
+    this.applicationContext = applicationContext;
+  }
+
+  public static String getJobInfoPath(String jobId) {
+    return jobId + "/" + JOB_FILENAME;
+  }
+
+  /**
+   * Creates job metadata, serializes to JSON and saves it to GCS.
+   *
+   * @param jobId
+   * @param countryCode
+   * @throws Exception
+   */
+  public void saveJobMetaData(String jobId, String countryCode) throws Exception {
+    try {
+      var jobInfoPath = getJobInfoPath(jobId);
+      var jobBlobId = BlobId.of(JOB_INFO_BUCKET_NAME, jobInfoPath);
+      var jobBlobInfo = BlobInfo.newBuilder(jobBlobId).setContentType("application/json").build();
+      var om = new ObjectMapper();
+      var jobMetadata = new JobMetadata(jobId, countryCode);
+      var json = om.writeValueAsString(jobMetadata);
+      logger.info("Saving job metadata: " + json);
+      storage.create(jobBlobInfo, json.getBytes());
+    } catch (Exception exc) {
+      logger.error("Error setting country code", exc);
+      throw exc;
+    }
+  }
+
+  /**
+   * Loads job metadata from GCS.
+   *
+   * @param jobId
+   * @return
+   */
+  public String getJobCountryCode(String jobId) {
```

Comment:
Maybe this should be named "getJobMetadata" and return JobMetadata directly?  For future flexibility.

Reply:
@bdferris-v2 It's really only saving the country code in a file that I just happened to leave the jobID in there for reference but it's never used.  Maybe I should change the object to JobCountryCode instead? We are already using "MetaData" in another context, it might be less confusing?

Reply:
I think there is a significant likelihood that we'll be adding more details to this metadata file in the future, not less.  So I'm ultimately in favor of keeping this general.  Understood that this is maybe not your problem :) but just calling out where I think this code is headed.

Reply:
Probably a similar issue with Javadoc @params.  The error should be listed at the end of the output.